### ISSUE TITLE
Android Auto Support

### ISSUE THREAD

**Issue by bioluks** [jarnedemeulemeester/findroid#715]
Hi,
Was shocked nobody mentioned this one. I know there's the official Jellyfin App, but it lacks stability on Android Auto (telling me libraries are empty all the time etc). Also see [this issue](https://github.com/jellyfin/jellyfin-android/issues/139) on Jellyfin-Android.

**Describe the solution you'd like**
Since this client already supports downloads and local playback (funnily still absent and will be for a while on the official client) I'm sure many would like the idea of playing movies/tv shows in the car without wasting their mobile data.

**Additional context**
Finamp is about to [support Android Auto](https://github.com/jmshrv/finamp/pull/578) as well (music focused Jellyfin player).

**Comment by jarnedemeulemeester** [comment]
My main priority right now is on Phone and TV support.
Once Findroid eventually add music support then Android Auto may be a good next target platform.

**Comment by AstroTheAstronaut** [comment]
Ugh... not to be the killjoy here but uhh... Streaming movies and shows in a car? While you drive?

**Comment by bioluks** [comment]
Ugh... uhmm... hmm:

1. Maybe it's not me who wants to watch the content
2. Could be that I'm parked, getting some rest and want to watch a tv show
3. Waiting for others in the car

The list of use cases go on. You could also visit [Fermata's GitHub](https://github.com/AndreyPavlenko/Fermata) and ask everybody in the issues there why they are using IPTV or YouTube in the car. I'm sure there are many who find themselves in these three situations mentioned above. The dev already gave his opinion on this.

**Comment by AstroTheAstronaut** [comment]
I get that, but:
_The core principle of Android for Cars is: Design for driving. That is, driving safely is the driver’s first responsibility. All design by car makers and app developers must reflect this priority.

App content and interactions should complement the driving experience while minimizing driver distraction. UI must be simplified to help keep the driver’s eyes on the road and hands on the wheel._

_Avoid displaying dynamic visual information unrelated to driving, such as videos and auto-scrolling text. Carefully consider the use of animations to ensure they aid the driver’s situational understanding._

These are all from Google's design foundations for Android Auto apps, and Google manually verifies each app before it allows it to be used in the Auto interface. So even if someone adds Android Auto support for video content, it's probably going to get blocked by Google.
Jellyfin also has an Android Auto app, however it's limited to music content only. If it's for music then yes, I want that too.
I know you can sideload apps, but most people aren't knowledgeable enough to do this, so developing a whole other ui that very few people are going to use is probably not a priority for anyone here. 
Also, to comment on one of your points (1). Are you honestly telling me that you won't get distracted at all by a literal show or movie playing exactly in your eyesight?

**Comment by bioluks** [comment]
It feels like big corpo software solutions these days are made by lawyers. Google just wants to stay out of legal trouble, I'm sure we will see _'dynamic visual information unrelated to driving'_ the next years. On most new Android Auto headunits you cannot drive unless you stop the car. In a very crowded city with arguably the worst drivers, you simply cannot park and enter your destination all the time. Imagine everyone did that, the traffic this would lead to.

I don't know if it's a priority for Findroid to stay in the Android Auto allowed apps whitelist, [even the best open-source E-Mail client was removed from Google Play, flagged as spyware a while ago](https://xdaforums.com/t/closed-app-5-0-fairemail-fully-featured-open-source-privacy-oriented-email-app.3824168/post-86908335). Unless you have some viral tweet or loud community you won't be able to get your falsely flagged app on the allowlist easily again.

Most people having their Jellyfin instance can easily sideload Findroid, this is nothing compared to setting the server up. For the instance members (mom, dad or other friends) they have you! Not everyone has the time but if you already gave them access to your instance you can take another step and quickly show them how to sideload it. If someone happens to find a piece of paper on the street with a user:password combination for a Jellyfin instance I could understand that.

Commenting on your comment, 100%. Eyesight though? I have seen various Android Auto setups, in the cars I drive they are not in my eyesight directly, at least not distracting. Haven't seen a car where the headunit is disturbingly distracting. Even if it is a bit, should be no issue (speaking for myself). Maybe an absurd example, but we are all human after all, it could be that you find yourself in a car accident because you got distracted by a fly. Level of distraction varies from person to person.

**Comment by AstroTheAstronaut** [comment]
Yes, most things nowadays are run by lawyers. Welcome to the modern world! Since we're on the topic of car stuff, isn't that what the auto industry has been for the last 10-15 years? Designing cars with accountants and lawyers and then asking the engineers for some extra help? It sucks but the best we can do is adapt to whatever we're have thrown at us.

I'm not talking about sideloading android apps. I'm talking about sideloading an app that you then use to _sideload_ (more like force) another app to the Android Auto interface. Why? Because Google will most likely block the app so you'll need to force it to work on Android Auto by using another app which, from what I can gather, costs money. That's way more than most people are able and, more importantly, willing to do.

**Comment by bioluks** [comment]
I mean... okay I guess. I meant it feels like the big corpo programmers, the employees themselves behave like lawyers, maybe my choice of words was not good enough. The facts you stated in the first paragraph aside from your irrelevant 'welcome to the modern world' should be well-known already.

About your information from your "information gathering"... You don't need a paid app for sideloaded Android Apps to show up in the headunit. You are probably talking about AAAD or AAStore (never used them), maybe it's not both of them but I remember some downloader (forgot the name) that used ads or wants money for unlimited downloads of Android Auto apps. `pm install -i "com.android.vending" -r android-auto-app.apk` is enough, along with enabled Unknown Sources enabled from AA settings, not sure about that since I always have it enabled. So no, nothing paid is needed to achieve this. [KingInstaller](https://github.com/fcaronte/KingInstaller) makes this process easier for many.

**Comment by AstroTheAstronaut** [comment]
> I mean... okay I guess. I meant it feels like the big corpo programmers, the employees themselves behave like lawyers, maybe my choice of words was not good enough. The facts you stated in the first paragraph aside from your irrelevant 'welcome to the modern world' should be well-known already.
> 
> About your information from your "information gathering"... You don't need a paid app for sideloaded Android Apps to show up in the headunit. You are probably talking about AAAD or AAStore (never used them), maybe it's not both of them but I remember some downloader (forgot the name) that used ads or wants money for unlimited downloads of Android Auto apps. `pm install -i "com.android.vending" -r android-auto-app.apk` is enough, along with enabled Unknown Sources enabled from AA settings, not sure about that since I always have it enabled. So no, nothing paid is needed to achieve this. [KingInstaller](https://github.com/fcaronte/KingInstaller) makes this process easier for many.

ADB though? That's not even "way more" than most people are willing to do. That's already "given up a while ago". If you need to work around this by using developer tools or other apps then it's not user friendly and most people won't do it, so they just won't use it. 
Findroid isn't necessarily an app made for tech savvy people. It has a simple user interface with very little complicated stuff going on. Adding a feature that requires running commands from a terminal is insanely complex and scary for the average person. Using an app is better but still more than most people are willing to do.
Again, I'll say what I said in the beginning. Id jarne (or someone else) adds this to Findroid then that's incredible and I'm happy for everyone.

**Comment by bioluks** [comment]
I don't know what you are trying to achieve. I asked if AA can be supported, the dev said in the future, you jumped in with an attitude I wasn't able to make sense of, welcomed me to the modern era and find every alternative solution I provided way too hard to implement. You are just saying what I meant from the beginning repeatedly:
> Id jarne (or someone else) adds this to Findroid then that's incredible and I'm happy for everyone.

Closing this issue as we are both contributing nothing on this topic. Anyone else _really_ interested about AA-Wireless can freely reopen, else the talk we did here is more for Reddit or GitHub Discussions.
